Refactor settings management by introducing a dedicated module

lbonaldo opened the pull request:
# Description

This PR refactors PowerGenome configuration management by creating a dedicated `settings.py` module. The proposal implements a factory pattern with a context manager for creation of scenario settings.

## Changes
- Added `settings.py` with a new `Settings` class and configuration utilities
- Moved some configuration functions from `util.py`
- Added `tests/settings_test.py` with test suite and moved settings tests from `tests/util_test.py`
- Updated imports across several existing files to use new Settings module

## Note: Breaking Changes
Some configuration functions have been moved from `util.py` to `settings.py`

lbonaldo committed:
Add new settings.py module with configuration utils

This commit introduces a new module containing a
`Settings` class and utility functions for
managing configuration parameters. Some methods
have been ported from the existing util module.

lbonaldo committed:
Update imports and settings module API calls

This commit updates all imports previously pointing
to the util module and replaces the settings getter
in the main function to align with the new Settings
module structure.

lbonaldo committed:
Add tests to new Settings module

Some tests have been ported from the existing
util_test module.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gschivley committed:
Sort model regions and create zone number mapping in load_settings function

gschivley commented:
@lbonaldo I think this design looks good. You've created the `Settings` object and a way to set the context for use throughout the rest of the code. 

The next step is to refactor code so that the `settings` dictionary doesn't get passed through, arguments are explicit with defaults of None, and values are pulled from the global settings if they aren't passed as args. Do you think that should be a separate PR?

gschivley committed:
Fix comment grammar, as requested by copilot review

gschivley committed:
Create auto_fill_settings decorator and add to functions

The decorator uses get_current_settings() to fill arguments with a default value of None that have a matching parameter in the settings dictionary.

gschivley committed:
Refactor network_max_reinforcement to use vectorized operations for calculating Line_Max_Reinforcement_MW

Avoids a bug with new pandas nullable float that can't be part of a max() boolean comparison

gschivley committed:
Merge branch 'pudl-data-only' into lb/store-settings

gschivley committed:
Fix fillna usage in add_cap_res_network and remove commented settings parameter in main function

gschivley committed:
Rename regional_aggregations to region_aggregations for consistency in test cases
